submissions/kikinovk/OOP_exercise/index.js

```diff
@@ -0,0 +1,108 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Being {
+  constructor(name, gender, saying) {
+
+     this.name = name;
+     this.gender = gender;
+     this.saying = saying;
+     this.friends = [];
+     this.species = '';
+  };
+  addFriends(arr) {
+     arr.forEach(item => this.friends.push(item));
+  };
+  description() {
+     return `${this.saying} my name is ${this.name}, I am a ${this.species}, I am ${this.gender}, ${this.friends.length > 0 ? `My friends is ${ this.friends.map(item => item.name).join(' and ')}` : 'I have not got friends'},`
+  }
+};
+
+class Animal extends Being {
+  constructor(name, gender, saying) {
+     super(name, gender, saying);
+     this.paws = 4;
+
+  };
+  description() {
+     return `${super.description()},  I have ${this.paws} paws`
+  }
+};
+
+class Human extends Being {
+  constructor(name, gender = 'male', saying) {
+     super(name, gender, saying);
+     this.legs = 2;
+     this.hands = 2;
+     this.species = 'human';
+
+  };
+  description() {
+     return `${super.description()}, I have ${this.hands} hands and ${this.legs} legs`
+  }
+};
+
+class Dog extends Animal {
+  constructor(name, gender, saying) {
+     super(name, gender, saying);
+     this.species = 'dog';
+  }
+};
+
+class Cat extends Animal {
+  constructor(name, gender, saying) {
+     super(name, gender, saying);
+     this.species = 'cat';
+  }
+};
+
+class SuperHero extends Human {
+  constructor(name, gender, saying, species = 'cat-woman') {
+     super(name, gender, saying);
+     this.species = species;
+  };
+}
+
+const dog = new Dog ('Charlie', 'male', 'woof-woof!');
+const cat = new Cat ('Taffy', 'female', 'meow!');
+const man = new Human ('Bruce', 'male', 'Hi!');
+const woman = new Human ('Emma', 'female', 'Hello!');
+const catWoman = new SuperHero ('Patience', 'female', cat.saying);
+
+
+dog.addFriends([man, woman]);
+cat.addFriends([man, woman, catWoman]);
+man.addFriends([woman, dog, cat, catWoman]);
+woman.addFriends([dog, cat, man]);
+
+const   inhabitants = [dog, cat, man, woman, catWoman];
+
+
+// // ======== OUTPUT ========
+
+inhabitants.forEach(obj => print(obj.description()));
```

Comment:
Every object has magic `toString` method that can be overloaded. It is called implicitly in string context. And `print` ensures string context. It is default implementation of `toString` that results in `[Object object]` in alerts, constructions like `.... ${obj}...` and older NodeJS engines.
Check [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString) for details.